src/Consolonia.PlatformSupport/CursesConsole.cs

```diff
@@ -608,107 +612,70 @@ private void HandleMouseInput(Curses.MouseEvent ev)
+                        RaiseMouseEvent(RawPointerEventType.MiddleButtonDown, point, null,
+                            rawInputModifiers | RawInputModifiers.MiddleMouseButton);
+                        break;
+                    case Curses.Event.Button2Released:
+                        RaiseMouseEvent(RawPointerEventType.MiddleButtonUp, point, null,
+                            rawInputModifiers | RawInputModifiers.MiddleMouseButton);
                         break;
                     case Curses.Event.Button3Pressed:
-                        _moveModifers = rawInputModifiers | RawInputModifiers.RightMouseButton;
-                        RaiseMouseEvent(RawPointerEventType.RightButtonDown, point, wheelDelta,
-                            rawInputModifiers | _moveModifers);
+                        RaiseMouseEvent(RawPointerEventType.RightButtonDown, point, null,
+                            rawInputModifiers | RawInputModifiers.RightMouseButton);
                         break;
-                    case Curses.Event.Button1Released:
-                        _moveModifers = RawInputModifiers.None;
-                        RaiseMouseEvent(RawPointerEventType.LeftButtonUp, point, wheelDelta,
-                            RawInputModifiers.None);
+                    case Curses.Event.Button3Released:
+                        RaiseMouseEvent(RawPointerEventType.RightButtonUp, point, null,
+                            rawInputModifiers | RawInputModifiers.RightMouseButton);
                         break;
-                    case Curses.Event.Button2Released:
-                        _moveModifers = RawInputModifiers.None;
-                        RaiseMouseEvent(RawPointerEventType.MiddleButtonUp, point, wheelDelta,
-                            RawInputModifiers.None);
+                    case Curses.Event.Button4Pressed:
+                        RaiseMouseEvent(RawPointerEventType.XButton1Down, point, null,
+                            rawInputModifiers | RawInputModifiers.XButton1MouseButton);
                         break;
-                    case Curses.Event.Button3Released:
                     case Curses.Event.Button4Released:
-                        _moveModifers = RawInputModifiers.None;
-                        RaiseMouseEvent(RawPointerEventType.RightButtonUp, point, wheelDelta,
-                            RawInputModifiers.None);
+                        RaiseMouseEvent(RawPointerEventType.XButton1Up, point, null,
+                            rawInputModifiers | RawInputModifiers.XButton1MouseButton);
                         break;
-
                     case Curses.Event.ReportMousePosition:
-                        RaiseMouseEvent(RawPointerEventType.Move, point, wheelDelta,
-                            rawInputModifiers | _moveModifers);
-                        break;
+                        if (point != _lastPoint)
```

Comment:
Why should we skip this?

Reply:
On my TTY I was getting mouse move events on the pixel level, which causes us to push a ton of events. For some reason the TTY mouse is already laggy (I would love input on how to make better) so I'm just trying to skip things that don't cause anything to happen.